[BUG] Scaladoc warnings emitted during build

The following Scaladoc warnings appear in recent builds:
```
/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuListUtils.scala:27: warning: Could not find any member to link for "scala.IllegalArgumentException".
  /**
  ^
```
and
```
/spark-rapids/shims/spark301/src/main/scala/org/apache/spark/sql/rapids/execution/python/shims/spark301/GpuArrowEvalPythonExec.scala:484: warning: Could not find any member to link for "GpuPythonUDF".
/**
^
```

--- a/pom.xml
+++ b/pom.xml
@@ -671,6 +671,7 @@
         <hive.storage.api.version>2.6.0</hive.storage.api.version>
         <protobuf.java.version>2.5.0</protobuf.java.version>
         <flatbuffers.java.version>1.11.0</flatbuffers.java.version>
+        <scala.local-lib.path>org/scala-lang/scala-library/${scala.version}/scala-library-${scala.version}.jar</scala.local-lib.path>
     </properties>
 
     <dependencyManagement>
@@ -901,6 +902,11 @@
                             <goals>
                                 <goal>doc-jar</goal>
                             </goals>
+                            <configuration>
+                                <args>
+                                    <arg>-doc-external-doc:${settings.localRepository}/${scala.local-lib.path}#https://scala-lang.org/api/${scala.version}/</arg>
+                                </args>
+                            </configuration>
                         </execution>
                     </executions>
                     <configuration>

